### DISCUSSION TITLE
API Routes Returning 404, Function Logs Empty (Even with Minimal Test Case)

### DISCUSSION THREAD

**Discussion by jntucehyd** [vercel/community#5906] [Help] [answered]
### Summary

Hello, 

I'm encountering a persistent issue with API routes on my Vercel deployments.  Even with extremely simplified test projects, the Vercel functions seem to not be executing at all. Here's what's happening:

**Behavior:** Accessing any API routes within my projects results in a 404 error.
**Empty Logs:** There is no corresponding log output generated in the Vercel dashboard's function logs, even when I include explicit console.log statements.

**Troubleshooting Steps Taken:**

I've worked with a helpful guide on this forum to try and isolate the problem. Here's what we've tried:

1. Simplified API Routes: Removed complex logic and even tested with the most basic response functions (e.g., returning a simple JSON object). The 404 behavior persists.
2. Extensive Logging: Added console.log statements at the start of my API route functions to ensure they're being reached. No logging output appears in Vercel logs.
3. Minimal Test Project: Created a brand-new Next.js project with the absolute minimum setup:
- A single API route file (pages/api/index.js) with basic logging and response.
- Deployed this to a new Vercel project.
- The same issue occurs – 404 errors and no function logging.
4. Vercel Status: Checked the Vercel status page, and there are no reported outages or issues.

**Code Example (Minimal Test Project):**

JavaScript
// pages/api/index.js
console.log('Minimal Test Function Running!'); 

export default (req, res) => {
    res.status(200).json({ message: 'Hello from the test API route!' });
}

**Questions:**

1. Has anyone else encountered similar behavior where Vercel functions don't seem to execute at all?
2. Are there any further configuration checks or troubleshooting steps I might be missing?
3. Could this be a temporary issue on Vercel's side?

Screenshots from my vercel project dashboards, logs etc. are attached

### Steps to Reproduce

![image](https://github.com/vercel/community/assets/20374299/63cba324-f5bf-4de3-83f8-c63e92325033)
![image](https://github.com/vercel/community/assets/20374299/bfb37e9c-67b3-4ad6-a1f7-e6f22ab38baf)
![image](https://github.com/vercel/community/assets/20374299/07ce3eee-2848-4d1e-b0fa-1b49d4cfa270)
![image](https://github.com/vercel/community/assets/20374299/e2857ec0-2230-4ac7-805d-d3c833d5287a)
![image](https://github.com/vercel/community/assets/20374299/2d64ed22-5f12-4269-9b7b-c3eb6c2182d7)

**Comment by amyegan** [comment, marked as answer]
Hi @jntucehyd. I wasn't able to reproduce the issue. Here's what I tried:
1. `npx create-next-app` to create a new starter project 
    <img width="691" alt="image" src="https://github.com/vercel/community/assets/8247105/676c5680-a6a3-40c2-8d3f-50606a268852">
2. Deploy starter project, visit `/api/hello` to check included starter function, see expected JSON `{"message":"Hello"}` output
3. Add `index.js` with a function similar to what you shared
    <img width="750" alt="image" src="https://github.com/vercel/community/assets/8247105/3dd439e0-1f8e-4d48-98eb-a3eef11078fa">
4. Deploy again, visit `/api` to check the new function, see expected JSON `{"message":"I am index.js"}` output and log 
    <img width="869" alt="image" src="https://github.com/vercel/community/assets/8247105/0621dd58-2781-4885-94ac-2f9aff4aa704">

If you're still having trouble with your serverless function, it may help to share a complete [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example). That way someone here may be able to dig into the code and recognize what's going wrong. 🙏
